optimum/gptq/quantizer.py

```diff
@@ -0,0 +1,629 @@
+
+        Args:
+            module (`nn.Module`):
+                Module to quantize
+            names (`List[str]`):
+                List of names of the module to quantize
+            name (`str`, defaults to `""`):
+                To keep track of the name of the current module
+        """
+        QuantLinear = dynamically_import_QuantLinear(
+            use_triton=False, desc_act=self.desc_act, group_size=self.group_size
+        )
+        if isinstance(module, QuantLinear):
+            return
+        for attr in dir(module):
+            layer = getattr(module, attr)
+            name1 = name + "." + attr if name != "" else attr
+            if name1 in names:
+                device = get_device(layer)
+                delattr(module, attr)
+                if isinstance(layer, nn.Linear):
+                    in_features = layer.in_features
+                    out_features = layer.out_features
+                elif isinstance(layer, nn.Conv2d):
+                    in_features = layer.in_channels
+                    out_features = layer.out_channels
+                elif isinstance(layer, Conv1D):
+                    in_features = layer.weight.shape[0]
+                    out_features = layer.weight.shape[1]
+                if not (self.desc_act) or self.group_size == -1:
+                    new_layer = QuantLinear(
+                        self.bits, self.group_size, in_features, out_features, True, use_cuda_fp16=self.use_cuda_fp16
+                    )
+                else:
+                    new_layer = QuantLinear(self.bits, self.group_size, in_features, out_features, True)
+                new_layer.device = device
+                setattr(module, attr, new_layer.to(device))
+        for name1, child in module.named_children():
+            self._replace_by_quant_layers(child, names, name + "." + name1 if name != "" else name1)
+
+    @torch.no_grad()
+    def quantize_model(self, model: nn.Module, tokenizer: Any, verbose: bool = False):
+        """
+        Quantizes the model using the dataset
+
+        Args:
+            model (`nn.Module`):
+                The model to quantize
+            tokenizer (`Any`):
+                The tokenizer to use in order to prepare the dataset. You can pass either:
+                    - A custom tokenizer object.
+                    - A string, the *model id* of a predefined tokenizer hosted inside a model repo on huggingface.co.
+                      Valid model ids can be located at the root-level, like `bert-base-uncased`, or namespaced under a
+                      user or organization name, like `dbmdz/bert-base-german-cased`.
+                    - A path to a *directory* containing vocabulary files required by the tokenizer, for instance saved
+                      using the [`~PreTrainedTokenizer.save_pretrained`] method, e.g., `./my_model_directory/`.
+            verbose (`bool`):
+                Whether to show the tqdm bar or not
```

Comment:
Why not show it by default and remove this? If it takes several hours I think it can be useful

Reply:
works for me !